v0/destinations/webengage/transform.js

```diff
@@ -0,0 +1,161 @@
+const { EventType } = require("../../../constants");
+const {
+  CONFIG_CATEGORIES,
+  MAPPING_CONFIG,
+  WEBENGAGE_IDENTIFY_EXCLUSION,
+  BASE_URL,
+  BASE_URL_IND
+} = require("./config");
+
+const {
+  defaultRequestConfig,
+  constructPayload,
+  defaultPostRequestConfig,
+  CustomError,
+  ErrorMessage,
+  extractCustomFields
+} = require("../../util");
+const moment = require("moment");
+const logger = require("../../../logger");
+
+const isValidTimestamp = timestamp => {
+  const re = /^(-?(?:[1-9][0-9]*)?[0-9]{4})-(1[0-2]|0[1-9])-(3[01]|0[1-9]|[12][0-9])T(2[0-3]|[01][0-9]):([0-5][0-9]):([0-5][0-9])(.[0-9]+)?(Z)?$/;
+  return re.test(String(timestamp));
+};
+
+const responseBuilder = (message, category, { Config }) => {
+  let payload = constructPayload(message, MAPPING_CONFIG[category.name]);
+  if (!payload) {
+    // fail-safety for developer error
+    throw new CustomError(ErrorMessage.FailedToConstructPayload, 400);
+  }
+  if (!payload.userId && !payload.anonymousId) {
+    throw new CustomError(
+      "[WEBENGAGE]: Either one of userId or anonymousId is mandatory.",
+      400
+    );
+  }
+  let baseUrl, endPoint;
+  const dataCenter = Config.dataCenter;
+  switch (dataCenter) {
+    case "ind":
+      baseUrl = BASE_URL_IND;
+      break;
+    default:
+      baseUrl = BASE_URL;
+  }
+
+  if (category.type === "identify") {
+    const customAttributes = {};
+    extractCustomFields(
+      message,
+      customAttributes,
+      ["context.traits", "traits"],
+      WEBENGAGE_IDENTIFY_EXCLUSION
+    );
+
+    payload = { ...payload, attributes: customAttributes };
+    endPoint = `${baseUrl}/${Config.licenseCode}/users`;
+  } else {
+    const eventTimeStamp = payload.eventTime;
+    if (eventTimeStamp) {
+      if (isValidTimestamp(eventTimeStamp)) {
+        payload.eventTime = moment(eventTimeStamp).format(
+          "YYYY-MM-DDThh:mm:sZZ"
+        );
+      } else {
+        logger.error("timestamp format must be ISO 8601.");
```

Comment:
Mention the exact format webengage is asking for.

Reply:
Yeah, done